### PR TITLE
MLflow UI Sync

### PR DESCRIPTION
**Pull request by daniellok-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/daniellok-db/mlflow/pull/11233?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11233/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11233
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

This PR syncs the latest UI changes, including multiple charting improvements.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

Showcase of some chart improvements:

1. Legend is added
2. Multiple metrics are supported in a single chart
3. Metrics can be grouped
4. Charts can be drag and dropped
5. Up to 100 runs are supported for line charts (though this isn't a good demo)

https://github.com/mlflow/mlflow/assets/148037680/8aaf86a5-e947-446f-b9b0-db177585ebbe

New experiment view and run page view:

https://github.com/mlflow/mlflow/assets/148037680/64adda88-e491-49bc-91e1-8e68b2ba7c5d

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

The MLflow UI has been updated with multiple new features:

- A new, updated run details page with new UI metric charts has been added. Please feel free to submit feedback on [this issue](https://github.com/mlflow/mlflow/issues/6348)!
- Charts now have a legend to more easily distinguish runs and metrics
- Charts in experiment and run details pages can be drag & dropped to reorder
- Experiment runs can be grouped by tags, parameters and datasets (grouped runs have aggregated metric values)
- Line charts can now display up to 100 runs
- Line charts now support multiple metrics in a single chart
- Improved experiment view state management (boosted UI performance)
- The current experiment view state can be shared using short, shareable URLs (click the “Share” button on the experiment page)

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by daniellok-db** [comment]
cc @hubertzub-db see [this link](https://github.com/mlflow/mlflow/pull/11233/files/dca515a53a431717beb5f9a3aeec334237ce4b54..ec83f761e8f9f9b65d94854ba0e887ce9acbb510) for the list of manual changes. In summary:

1. Flip the flags for Deep Learning UI Phases 1 - 3 (should I leave #3 unflipped for now?)
2. [Restore the info tooltip and feedback form URL on the experiment view](https://github.com/mlflow/mlflow/pull/11233/files/dca515a53a431717beb5f9a3aeec334237ce4b54..ec83f761e8f9f9b65d94854ba0e887ce9acbb510#diff-46741c37ebf5fef838aeba13abaf99941df690bc63d049c06a6a02ba89f3bc64R81). Without this change, users can't get the ID or path of their experiments.
3. [Don't attempt to fetch an empty metric key](https://github.com/mlflow/mlflow/pull/11233/files/dca515a53a431717beb5f9a3aeec334237ce4b54..ec83f761e8f9f9b65d94854ba0e887ce9acbb510#diff-31c155b6155ae1d90c73e958ef5f5cbf72f03838a524829d58793f185e454756R53) in `getSampledMetricHistoryBulkAction`. I'm not sure under what circumstances an empty metric key can be passed to this function, but it did happen in my testing and caused the server to throw an exception.
4. Fixed a bunch of tests related to the flag-flipping

File: mlflow/server/js/OWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+per-file .enzymemigrationignore=*
```
Comment (hubertzub-db) [inline mlflow/server/js/OWNERS:1]: should we ignore OWNERS file in copybara settings?

**Review by hubertzub-db — APPROVED** [review]
lg!

File: mlflow/server/js/OWNERS (hunk 1 above)
Comment (daniellok-db) [reply mlflow/server/js/OWNERS:1]: sounds good to me, how do i do that? should i also remove this file from the PR?

**Pull request merged**